### PR TITLE
feat: Make closable dialog emit event on close #790

### LINKED ISSUE #790: Dialog is not automatically cleared when closable is True
See https://github.com/h2oai/wave/discussions/789 via @vopani 

Setting `closable = True` should clear the dialog from the meta_card's state. Currently, the dialog is marked as dismissed, but it continues to persist in the meta_card.

### PR DESCRIPTION
**Pull request by mturoci**
This PR allows event fired on closing of dialog if `closable` and `name` is specified.

Closes #790

### CODE REVIEW

File: ui/src/dialog.tsx [hunk 1]
```diff
@@ -38,15 +38,19 @@ export interface Dialog {
...
   primary?: B
+  /** An identifying name for this component. */
+  name?: Id
 }
 
 export default bond(() => {
   const
     onDismiss = () => {
+      const { closable, name } = dialogB() || {}
+      if (closable && name) {
```
Comment (lo5) [inline ui/src/dialog.tsx:49]: Raise event only if subscribed.

**Review by lo5 — CHANGES_REQUESTED** [review]

**Comment by lo5** [comment]
Additionally, needs a screenshot for the new example.

File: ui/src/dialog.tsx [hunk 2]
```diff
@@ -38,15 +38,19 @@ export interface Dialog {
   blocking?: B
   /** Dialog with large header banner, mutually exclusive with `closable` prop. Defaults to false. */
   primary?: B
+  /** An identifying name for this component. */
+  name?: Id
```
[the remark this answers is not in the document]
Comment (mturoci) [reply ui/src/dialog.tsx:42]: Done.

File: ui/src/dialog.tsx (hunk 1 above)
Comment (mturoci) [reply ui/src/dialog.tsx:51]: Done.

File: py/examples/meta_dialog_closable.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+@app('/demo')
+async def serve(q: Q):
+    if not q.client.initialized:
+        q.page['meta'] = ui.meta_card(box='')
+        q.page['example'] = ui.form_card(box='1 1 2 1', items=[
+            ui.button(name='show_dialog', label='Open dialog', primary=True)
+        ])
+        q.client.initialized = True
+    if q.args.show_dialog:
+        q.page['meta'].dialog = ui.dialog(title='Order Donuts', name='dialog', closable=True, items=[
```
[the remark this answers is not in the document]
Comment (mturoci) [reply py/examples/meta_dialog_closable.py:16]: Done.

**Review by lo5 — DISMISSED** [review]

**Review by lo5 — APPROVED** [review]

**Pull request merged**